swift/downgrades/58e9efb426b5b2b322a64c9dab988b02578e0f04/downgrade.ql

```diff
@@ -0,0 +1,11 @@
+class Element extends @element {
+  string toString() { none() }
+}
+
+query predicate new_unspecified_elements(Element e, string property, string error) {
+  unspecified_elements(e, property, error)
+  or
+  error = "Macro declarations removed during database downgrade. Please update your CodeQL code." and
+  property = "" and
+  macro_decls(e, _)
```

Comment:
aren't `macro_roles` missing here?

Reply:
I think they can be safely deleted as they are only referenced by macro decls?

Reply:
you might be right actually